Undeclared bool type in magma header file

# Description

Compilation of a code including the MAGMA header file (v1 or v2) fails with the oneAPI compiler (version 2024.2) due to missing declaration of the `bool` type.

# To reproduce

Compile `test.c` with following content
```C
#include <magma_v2.h>

void dummy() {
}
```
with
```
icx -c test.c
```

# Obtained result

```
In file included from test.c:1:
In file included from /opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_v2.h:31:
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_z.h:63:1: error: unknown type name 'bool'
   63 | bool magma_zgetrf_gpu_recommend_cpu(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_z.h:64:1: error: unknown type name 'bool'
   64 | bool magma_zgetrf_native_recommend_notrans(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
In file included from test.c:1:
In file included from /opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_v2.h:32:
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_c.h:63:1: error: unknown type name 'bool'
   63 | bool magma_cgetrf_gpu_recommend_cpu(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_c.h:64:1: error: unknown type name 'bool'
   64 | bool magma_cgetrf_native_recommend_notrans(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
In file included from test.c:1:
In file included from /opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_v2.h:33:
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_d.h:63:1: error: unknown type name 'bool'
   63 | bool magma_dgetrf_gpu_recommend_cpu(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_d.h:64:1: error: unknown type name 'bool'
   64 | bool magma_dgetrf_native_recommend_notrans(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
In file included from test.c:1:
In file included from /opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_v2.h:34:
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_s.h:63:1: error: unknown type name 'bool'
   63 | bool magma_sgetrf_gpu_recommend_cpu(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
/opt/packager/manual/opt/magma/2.9_intel-2024.2_cuda-12.4/include/magma_s.h:64:1: error: unknown type name 'bool'
   64 | bool magma_sgetrf_native_recommend_notrans(magma_int_t m, magma_int_t n, magma_int_t nb);
      | ^
8 errors generated.
```

# Expected result

Code should compile without error message

# Workaround

Including `<stdbool.h>` before the inclusion of `magma_v2.h` on the user side works around the problem, but I think this should happen on MAGMA's side.

--- a/include/magma_types.h
+++ b/include/magma_types.h
@@ -16,6 +16,9 @@
 #include <stdint.h>
 #include <assert.h>
 
+#ifndef __cplusplus
+#include<stdbool.h>
+#endif
 
 // for backwards compatability
 #ifdef HAVE_clAmdBlas

